frontend/src/metabase/admin/databases/components/DatabaseEditApp/Sidebar/Sidebar.tsx

```diff
@@ -0,0 +1,169 @@
+import React, { useCallback, useRef } from "react";
+import { t } from "ttag";
+
+import Button from "metabase/core/components/Button";
+import ActionButton from "metabase/components/ActionButton";
+import ConfirmContent from "metabase/components/ConfirmContent";
+import ModalWithTrigger from "metabase/components/ModalWithTrigger";
+
+import { isSyncCompleted } from "metabase/lib/syncing";
+import DeleteDatabaseModal from "metabase/admin/databases/components/DeleteDatabaseModal.jsx";
+
+import type { DatabaseId } from "metabase-types/api";
+import type Database from "metabase-lib/metadata/Database";
+
+import ModelCachingControl from "./ModelCachingControl";
+import { SidebarRoot, SidebarContent, SidebarGroup } from "./Sidebar.styled";
+
+interface DatabaseEditAppSidebarProps {
+  database: Database;
+  isAdmin: boolean;
+  isModelPersistenceEnabled: boolean;
+  updateDatabase: (database: Database) => void;
+  syncDatabaseSchema: (databaseId: DatabaseId) => void;
+  dismissSyncSpinner: (databaseId: DatabaseId) => void;
+  rescanDatabaseFields: (databaseId: DatabaseId) => void;
+  discardSavedFieldValues: (databaseId: DatabaseId) => void;
+  deleteDatabase: (databaseId: DatabaseId, isDetailView: boolean) => void;
+}
+
+const DatabaseEditAppSidebar = ({
+  database,
+  deleteDatabase,
+  syncDatabaseSchema,
+  dismissSyncSpinner,
+  rescanDatabaseFields,
+  discardSavedFieldValues,
+  isAdmin,
+  isModelPersistenceEnabled,
+}: DatabaseEditAppSidebarProps) => {
+  const discardSavedFieldValuesModal = useRef<any>();
+  const deleteDatabaseModal = useRef<any>();
+
+  const isSynced = isSyncCompleted(database);
+  const hasModelCachingSection =
+    isModelPersistenceEnabled && database.supportsPersistence();
+
+  const handleSyncDatabaseSchema = useCallback(
```

Comment:
are all these `useCallbacks` really necessary? are the props to this component likely to change frequently or at all? I think the overhead of creating them far outweighs their benefits here. Also, would be a bit easier to read without them.

Reply:
Pardon my React performance knowledge, but isn't it still less overhead than creating a new function on every re-render? Unfortunately, given our `*App` patter where a top-level component subscribes to the whole Redux state tree and connects all the available module actions, re-renders happen much more frequently through the whole app

Reply:
I did some very high-tech `console.log('render')` debugging, and didn't observe any re-renders of this component after the 3 initial renders. What in the redux store would change the props of this component?

Reply:
Haven't tried that, but I suppose any connection form edit would trigger a rerender since we pass the whole `database` object to the sidebar. Let me explore this a little bit further

Reply:
now that our forms framework is improved, I think we only update the database object in redux when we save the form, not on every keystroke like before.

I don't want to make much ado about nothing, this is a very small performance optimization at best.

Reply:
Oh right! Can't believe the `redux-form` era is over